pyjob/lsf.py

```diff
@@ -55,11 +55,13 @@ def info(self):
         else:
             return {'job_number': self.pid, 'status': 'Running'}
 
-    def close(self):
-        """Close this :obj:`~pyjob.lsf.LoadSharingFacilityTask` after completion"""
-        self.wait()
-        if self.cleanup_files and self.runscript is not None:
-            self.runscript.cleanup()
+    def _check_requirements(self):
+        """Check if the requirements for task execution are met"""
+
+        try:
+            cexec(['bjobs'])
+        except PyJobExecutableNotFoundError:
+            raise PyJobError('Cannot find LSF. Please ensure this is the correct platform to run your task!')
```

Comment:
I think we should put something like this in the base class, i.e.

```
@staticmethod
def _ensure_exec_available(exec):
    try:
        cexec([exec])
    except PyJobExecutableNotFoundError:
        raise PyJobError('Cannot find executable {}. Please ensure environment is set up correctly.'.format(exec)) 
```

then in all of your base classes you can simply add to `_check_requirements` a call to `self._ensure_exec_available('bjobs')`